src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+                        await StateManager.GetOrAddAsync<IReliableDictionary<string, DateTime>>("update");
+                    try
+                    {
+                        using (ITransaction tx = StateManager.CreateTransaction())
+                        {
+                            var previousTransactionDateTime = await update.GetOrAddAsync(
+                             tx,
+                             "update",
+                             DateTime.Now
+                             );
+                            var updatedTime = await CheckForError(previousTransactionDateTime, DependencyUpdateErrorProcessorOptions.GithubUrl);
+                            await update.TryUpdateAsync(
+                                tx,
+                                "update",
+                                updatedTime.UtcDateTime,
+                                previousTransactionDateTime
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, "Processing update dependency messages");
+                    }
+                }
+            }
+            else
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+            }
+        }
+
+        public async Task<DateTimeOffset> CheckForError(DateTimeOffset previousTransactionDateTime , string whereToCreateIssue)
+        {
+            List<RepositoryBranchUpdateHistoryEntry> list = (from repo in Context.RepositoryBranchUpdateHistory
+                                                             where repo.Success == false
+                                                             where repo.Timestamp > previousTransactionDateTime.UtcDateTime
+                                                             orderby repo.Timestamp ascending
+                                                             select repo).ToList();
+            if (!list.Any())
+            {
+                return previousTransactionDateTime;
+            }
+
+            Logger.LogInformation($"Going to create the github issue");
+            try
+            {
+                await CreateGitHubIssueAsync(previousTransactionDateTime.UtcDateTime,  list[0].Repository, list[0].Branch, 
+                    list[0].Action, list[0].ErrorMessage, list[0].Method , whereToCreateIssue);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Could not create a github issue for the error: " + list[0].ErrorMessage);
+            }
+            previousTransactionDateTime = list[0].Timestamp;
+            return previousTransactionDateTime;
+        }
+
+        public async Task<bool> SearchCreatedIssueAsync(DateTime transactionDateTime , string errorMessage)
```

Comment:
Would it make sense to add some FindIssue method in the GitHubClient https://github.com/dotnet/arcade-services/blob/master/src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs and use that in this method?

Reply:
We have our own implementation of GitHubClient that isn't just Octokit?  Is there something it doesn't better/different?

Reply:
It's an implementation for [IGitRepo](https://github.com/dotnet/arcade-services/blob/master/src/Microsoft.DotNet.Darc/src/DarcLib/IGitRepo.cs). It mostly wraps octokit calls, and does direct API calls for some operations that were either not entirely supported by octokit, or for things that we wrote before we started using octokit.

Reply:
Hrm... having to go write some method on some interface, then implement it in some class seems... heavy-handed.  Can't we just use octokit directly where it works, and just have extensions to it, rather than wrappers?  Having to go add "also add this label to the issue", and then implementing the entire search layer is going to be a lot of work for no gain.

Reply:
Sure, it was mostly since we already have an octokit client handy in that class, and FindIssue seemed like a nice helper, but if we add the method to that interface, we'd need to add non-implemented versions of the method in the Azdo implementation, so probably not much gain.